### PR TITLE
FAI-11383: CircleCI Memory fix

### PR DESCRIPTION
**Pull request by willmarks**
## Description

Currently, the CircleCI converter stores all test executions it sees so that it can aggregate all of the test case result stats based off of test records it converts. This has caused issues where the destination will run out of memory. However, we can instead only worry about on test execution at a time because the source emits records related to a single test execution in order. Since we can assume that all test case results related to a test execution will be processed in order before moving onto the next test execution, we can aggregate one TestExecution at a time until we  see the first record that is from a new TestExecution at which point we send in the TestExecution and start aggregating the next. The final TestExecution needs to be sent in the onProcessComplete logic

## Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change

### CODE REVIEW

**Review by tovbinm — APPROVED** [review]

File: destinations/airbyte-faros-destination/src/converters/circleci/tests.ts [hunk 1]
```diff
@@ -141,22 +141,30 @@ export class Tests extends CircleCIConverter {
...
+          test.pipeline_vcs,
+          test.project_slug
+        ),
+      };
+      // Reset test case results set
+      this.testCaseResults.clear();
     } else {
-      te.testCaseResultsStats.unknown += 1;
+      if (testCaseResultStatus.category === 'Failure') {
+        this.currentTestExecution.status = {category: 'Failure', detail: null};
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/circleci/tests.ts:155]: why is this necessary?

Comment (willmarks) [reply destinations/airbyte-faros-destination/src/converters/circleci/tests.ts:155]: Its cause as soon as we see one failure the whole test execution should be marked as failed. Also, I am just copying what we had already.
